Remotion/ObjectBinding/Validation.UnitTests/PropertyValidatorToBusinessObjectPropertyConstraintConverterTest.cs

```diff
@@ -15,17 +15,82 @@
 {
   [TestFixture]
   public class PropertyValidatorToBusinessObjectPropertyConstraintConverterTest
   {
     [Test]
-    [Ignore("TODO RM-5906")]
-    public void Convert ()
+    public void Convert_WithDifferentMaxLengthValidators_ContainsLowerMaxLength ()
     {
+      var validators = new []
+                       {
+                         new MaximumLengthValidator(7, new InvariantValidationMessage("Max length is 7")),
+                         new MaximumLengthValidator(5, new InvariantValidationMessage("Max length is 5"))
+                       };
+
+      var converter = new PropertyValidatorToBusinessObjectPropertyConstraintConverter();
+
+      var constraints = converter.Convert(validators).ToArray();
+
+      Assert.That(constraints.Length, Is.EqualTo(1));
+      Assert.That(constraints[0], Is.InstanceOf<BusinessObjectPropertyValueLengthConstraint>());
+      Assert.That(((BusinessObjectPropertyValueLengthConstraint)constraints[0]).MaxLength, Is.EqualTo(5));
+    }
+
+    [Test]
+    public void Convert_WithoutValidators_ContainsNoConstraints ()
+    {
+      var validators = Array.Empty<IPropertyValidator>();
+
+      var converter = new PropertyValidatorToBusinessObjectPropertyConstraintConverter();
+
+      var constraints = converter.Convert(validators).ToArray();
+
+      Assert.That(constraints, Is.Empty);
+    }
+
+    [Test]
+    public void Convert_WithRequiredValidators_ContainsRequiredConstraint ()
+    {
+      var validators = new []
+                       {
+                         new NotNullValidator(new InvariantValidationMessage("Required")),
+                         new NotNullValidator(new InvariantValidationMessage("Required number 2")),
+                       };
+
+      var converter = new PropertyValidatorToBusinessObjectPropertyConstraintConverter();
+
+      var constraints = converter.Convert(validators).ToArray();
+
+      Assert.That(constraints.Length, Is.EqualTo(1));
+      Assert.That(constraints[0], Is.InstanceOf<BusinessObjectPropertyValueRequiredConstraint>());
+    }
+
+    [Test]
+    public void Convert_WithRequiredAndLengthValidators_ContainsRequiredAndLengthConstraint ()
+    {
+      var validators = new IPropertyValidator[]
+                       {
+                         new NotNullValidator(new InvariantValidationMessage("Required")),
+                         new NotNullValidator(new InvariantValidationMessage("Required number 2")),
+                         new MaximumLengthValidator(5, new InvariantValidationMessage("Max length is 5"))
+                       };
+
+      var converter = new PropertyValidatorToBusinessObjectPropertyConstraintConverter();
+
+      var constraints = converter.Convert(validators).ToArray();
+
+      Assert.That(constraints.Length, Is.EqualTo(2));
+      Assert.That(constraints,
+          Has.Some.InstanceOf<BusinessObjectPropertyValueLengthConstraint>()
+              .And.Some.InstanceOf<BusinessObjectPropertyValueRequiredConstraint>());
```

Comment:
I do like this. Never seen it before. But I like it :)

Reply:
Was done & researched by @KuJo7, I simply put a comment there on how to do it better, he implemented the suggestion.